src/execution_plan/ops/op_merge.c

```diff
@@ -47,14 +47,13 @@ static void _UpdateProperty(Record r, GraphEntity *ge, EntityUpdateEvalCtx *upda
 }
 
 // Perform all ON MATCH updates for all matched records.
-static void _UpdateProperties(OpMerge *op, Record *records) {
+static void _UpdateProperties(OpMerge *op, Record *records, uint record_count) {
 	GraphContext *gc = QueryCtx_GetGraphCtx();
 	uint update_count = array_len(op->on_match);
-	uint record_count = array_len(records);
 
-	// Lock everything.
-	if(!QueryCtx_LockForCommit()) return;
-	{
+	if(op->on_match && update_count && record_count) {
+		// Lock everything.
+		if(!QueryCtx_LockForCommit()) return;
```

Comment:
`QueryCtx_LockForCommit` is always performed at run-time, which means an exception handler is setup,
there's no need to check for a return value in this case, as if `QueryCtx_LockForCommit` returns false it would jump to the exception handler, I believe this if condition can be reduced to just `QueryCtx_LockForCommit();` as such for all other places calling `QueryCtx_LockForCommit` besides maybe index deletion/creation.

Reply:
done